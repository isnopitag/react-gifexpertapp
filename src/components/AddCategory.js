import React, { useState } from "react";
import PropTypes from "prop-types";

export const AddCategory = ({ setCategories }) => {
  const [inputValue, setInputValue] = useState("");

  const handeInputChange = (e) => {
    setInputValue(e.target.value);
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    console.log('Handle Submit', inputValue);
    if (inputValue.trim() === "conejinga") {
      setCategories((c) => ["bunny", ...c]);
      alert("Te quiero mucho mi conejita 🐇");
      setInputValue("");
    } else {
      if (inputValue.trim().length > 2) {
        setCategories((c) => [inputValue, ...c]);
        setInputValue("");
      }
    }
  };
  return (
    <form onSubmit={handleSubmit}>

      <p>{inputValue}</p>
      <input type="text" value={inputValue} onChange={handeInputChange} />
    </form>
  );
};

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired,
};
