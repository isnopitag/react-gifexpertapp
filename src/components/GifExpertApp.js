import React, { useState } from "react";
import AddCategory from "./AddCategory.js";
import { GifGrid } from "./GifGrid.js";

const GifExpertApp = () => {
  const [categories, setCategories] = useState(["evangelion"]);

  return (
    <>
      <h2>GifExpertApp</h2>
      <hr />
      <AddCategory setCategories={setCategories} />
      <ol>
        {categories.map((category) => (
          <GifGrid 
            category={category}
            key={category}
            />
        ))}
      </ol>
    </>
  );
};

export default GifExpertApp;
