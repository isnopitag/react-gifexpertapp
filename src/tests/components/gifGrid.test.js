import React from "react";
import { GifGrid } from "../../components/GifGrid";
import { shallow } from "enzyme";
import { useFetchGifs } from "../../hooks/useFetchGifs";
jest.mock("../../hooks/useFetchGifs");

describe("Pruebas en GifGrid", () => {
  const category = "evangelion";

  test("debe de mostrar <GifGrid/>", () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    });
    const wrapper = shallow(<GifGrid category={category} />);

    expect(wrapper).toMatchSnapshot();
  });

  test("debe de mostrar items cuando se cargan imagenes useFetchGifs", () => {
    const gifs = [
      {
        id: "abc",
        url: "https://localhost/algo.jpg",
        title: "Cualquier cosa",
      },
      {
        id: "abd",
        url: "https://localhost/algo.jpg",
        title: "Cualquier cosa",
      },
    ];
    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false,
    });

    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchInlineSnapshot(`
      <Fragment>
        <h3
          className="animate__animated animate__flash animate__delay-1s "
        >
          evangelion
        </h3>
        <div
          className="card-grid"
        >
          <GifGridItem
            id="abc"
            key="abc"
            title="Cualquier cosa"
            url="https://localhost/algo.jpg"
          />
          <GifGridItem
            id="abd"
            key="abd"
            title="Cualquier cosa"
            url="https://localhost/algo.jpg"
          />
        </div>
      </Fragment>
    `);
    expect(wrapper.find("p").exists()).toBe(false);
    expect(wrapper.find("GifGridItem").length).toBe(gifs.length);
  });
});
