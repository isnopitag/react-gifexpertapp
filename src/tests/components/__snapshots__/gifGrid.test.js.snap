// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifGrid debe de mostrar <GifGrid/> 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__flash animate__delay-1s "
  >
    evangelion
  </h3>
  <p
    className="animate__animated animate__flash "
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
